src/OrbitBase/Logging.cpp

```diff
@@ -14,6 +14,24 @@
 
 static absl::Mutex log_file_mutex(absl::kConstInit);
 std::ofstream log_file;
+const int kLogFileLifetimeMinutes = 60 * 24 * 7;  // one week
+
+void RemoveOldLogFiles(const std::filesystem::path& log_dir) {
+  auto now = std::filesystem::file_time_type::clock::now();
+  for (const auto& log_file_path : std::filesystem::recursive_directory_iterator(log_dir)) {
+    std::chrono::minutes file_lifetime_minutes = std::chrono::duration_cast<std::chrono::minutes>(
+        now - std::filesystem::last_write_time(log_file_path));
+    try {
+      if (file_lifetime_minutes.count() > kLogFileLifetimeMinutes) {
+        std::filesystem::remove(log_file_path);
```

Comment:
std::filesystem functions always have noexcept couterpart that sets std::error_code. Please use them instead - we try to keep the code exception-free.

Reply:
Changed accordingly. Thanks for pointing out this!